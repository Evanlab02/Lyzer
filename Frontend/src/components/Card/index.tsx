import CardBody from "./CardBody";
import CardHeading from "./CardHeading";
import { PropsWithChildren } from "react";
import "./styles/index.scss";

export default function Card(props: PropsWithChildren) {
    const { children } = props;

    return(
        <div className="lyzer-card">
            {children}
        </div>
    );
}

export { CardBody, CardHeading };
