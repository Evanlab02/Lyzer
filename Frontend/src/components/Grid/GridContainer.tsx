import { PropsWithChildren } from "react";
import "./styles/GridContainer.scss";

export default function GridContainer(props: Readonly<PropsWithChildren>) {
	const { children } = props;

	return (
		<div className="flex-grid-container">
			{children}
		</div>
	);
}