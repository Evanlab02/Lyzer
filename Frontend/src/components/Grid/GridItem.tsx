import { PropsWithChildren } from "react";

export interface FlexGridItemProps extends PropsWithChildren {
    xs?: number
    sm?: number
    md?: number
    lg?: number
    xl?: number
    xxl?: number
}

export default function GridItem(props: Readonly<FlexGridItemProps>) {
	const { children, xs, sm, md, lg, xl, xxl } = props;

	const classes = [
		xs && `flex-grid-col-xs-${xs.toString()}`,
		sm && `flex-grid-col-sm-${sm.toString()}`,
		md && `flex-grid-col-md-${md.toString()}`,
		lg && `flex-grid-col-lg-${lg.toString()}`,
		xl && `flex-grid-col-xl-${xl.toString()}`,
		xxl && `flex-grid-col-xxl-${xxl.toString()}`
	].filter(Boolean).join(" ");

	return (
		<div className={classes}>
			{children}
		</div>
	);
}
