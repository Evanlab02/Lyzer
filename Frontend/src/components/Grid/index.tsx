export { default as GridContainer } from "./GridContainer";
export { default as GridItem } from "./GridItem";
