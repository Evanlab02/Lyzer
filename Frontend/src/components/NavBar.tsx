import { Menu, Moon, Sun } from "lucide-react";
import { useTheme } from "../hooks/useTheme";
import "../styles/navbar.scss";

type NavBarProps = {
	onMenuClick: (event: React.MouseEvent<SVGElement>) => void;
}

export default function NavBar({ onMenuClick}: NavBarProps) {
	const {isDarkMode, toggleTheme} = useTheme();

	return (
		<div className="navbar">
			<Menu 
				color={isDarkMode ? "white" : "black"} 
				size={28} onClick={onMenuClick} 
				className="icon menu"
				id="navbar-menu"
			/>
			<div>
				â€‹ðŸ‡±â€‹â€‹ðŸ‡¾â€‹â€‹ðŸ‡¿â€‹â€‹ðŸ‡ªâ€‹â€‹ðŸ‡·â€‹
			</div>
			{
				isDarkMode ? (
					<Moon
						id="theme-toggle-dark"
						className="icon theme-toggle"
						color="white"
						size={28}
						onClick={() => toggleTheme()}
					/>
				) : (
					<Sun
						id="theme-toggle-light"
						className="icon theme-toggle"
						color="black"
						size={28}
						onClick={() => toggleTheme()}
					/>
				)
			}
		</div>
	);
}