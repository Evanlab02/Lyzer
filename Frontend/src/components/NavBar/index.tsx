import { Menu, Moon, Sun } from "lucide-react";
import { useTheme } from "../../hooks/useTheme";
import "./styles/index.scss";

interface NavBarProps {
	testId?: string;
	menuToggleTestId?: string;
	themeToggleTestId?: string;
	onMenuClick: (event: React.MouseEvent<SVGElement>) => void;
}

export default function NavBar(props: Readonly<NavBarProps>) {
	const {
		testId,
		menuToggleTestId,
		themeToggleTestId,
		onMenuClick
	} = props;
	
	const {
		isDarkMode, 
		toggleTheme
	} = useTheme();

	return (
		<div className="navbar" data-testid={testId}>
			<Menu 
				color={isDarkMode ? "white" : "black"} 
				size={28} 
				onClick={onMenuClick} 
				className="icon menu"
				id="navbar-menu"
				data-testid={menuToggleTestId}
			/>
			<div>L Y Z E R</div>
			{
				isDarkMode ? 
					<Moon
						id="theme-toggle-dark"
						className="icon theme-toggle"
						color="white"
						size={28}
						data-testid={themeToggleTestId}
						onClick={() => { toggleTheme(); }}
					/>
					:
					<Sun
						id="theme-toggle-light"
						className="icon theme-toggle"
						color="black"
						size={28}
						data-testid={themeToggleTestId}
						onClick={() => { toggleTheme(); }}
					/>
				
			}
		</div>
	);
}