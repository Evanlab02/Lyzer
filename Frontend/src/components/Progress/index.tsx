import { useMemo } from "react";
import "./styles/index.scss";

export interface ProgressProps {
    value?: number;
    max?: number;
    useProgressColour?: boolean;
    testId?: string;
}

export default function Progress(props: Readonly<ProgressProps>) {
	const { 
		value = 0, 
		max = 100,
		useProgressColour = false,
		testId
	} = props;

	const progressClassName = useMemo(() => {
		const progressPercentage = (value / max) * 100;
		if (useProgressColour && progressPercentage == 100) {
			return "lyzer-progress-complete";
		} else if (useProgressColour && 80 <= progressPercentage && progressPercentage < 100) {
			return "lyzer-progress-almost";
		} else if (useProgressColour && progressPercentage < 80) {
			return "lyzer-progress-incomplete";
		}
	}, [value, max, useProgressColour]);

	return (
		<progress
			className={`lyzer-progress ${progressClassName ?? ""}`}
			value={value}
			max={max}
			data-testid={testId}
		/>
	);
}