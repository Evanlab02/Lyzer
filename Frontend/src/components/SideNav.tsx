import { ArrowLeftToLine } from "lucide-react";
import "../styles/sideNav.scss"
import { useTheme } from "../hooks/useTheme";
import React, { useEffect, useRef } from "react";
import { ROUTES } from "../consts/routes";
import { Link, useLocation } from "react-router-dom";

type SideNavProps = {
	sideNavOpen: boolean;
	onCloseClick: () => void;
	onSideNavBlur: React.FocusEventHandler<SVGSVGElement>;
}

export default function SideMenu({ sideNavOpen, onCloseClick, onSideNavBlur}: SideNavProps) {

	const {isDarkMode} = useTheme();

	const sideNavRef = useRef<HTMLDivElement>(null);

	const location = useLocation();

	useEffect(() => {
		function handleOutsideClick(event: MouseEvent) {
			if (sideNavRef.current && !sideNavRef.current.contains(event.target as Node)) {
				onCloseClick();
			}
		}

		if (sideNavOpen) {
			document.addEventListener("mousedown", handleOutsideClick);
		}

		return () => {
				document.removeEventListener("mousedown", handleOutsideClick);
		};
	}, [sideNavOpen, onCloseClick]);

	const isSelectedRoute: (route: string) => boolean = (route: string) => {
		return location.pathname === route;
	};

	return (
		<div id="side-menu" className={"side-nav " + (sideNavOpen ? "open" : "closed")} ref={sideNavRef}>
			<ArrowLeftToLine 
				onClick={onCloseClick}
				onBlur={onSideNavBlur}
				color={isDarkMode ? "white" : "black"}
				className="closeIcon"
				size={28}
			/>
			<div id="links" className="links">
				{/* Check if active link, apply active style
				Create hover style
				Add logo to top of side menu */}
				{Object.values(ROUTES).map((item, index) => (
					<Link 
						key={"linkContainer" + index} 
						className={"link-container " + (isSelectedRoute(item.route) ? 'selected' : '')}
						to={item.route}
					>
						{item.icon}
						<div key={"linkName" + index}>
							{item.name.toUpperCase()}
						</div>
					</Link>
				))}
			</div>
		</div>
	)
}