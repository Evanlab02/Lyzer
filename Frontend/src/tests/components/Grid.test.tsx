import { expect, it } from "vitest";
import { render } from "@testing-library/react";
import { GridContainer, GridItem } from "../../components/Grid";

it("Grid container renders correctly", () => {
	const { getByTestId } = render(
		<GridContainer testId="lyzer-grid-container">
			Grid Container
		</GridContainer>
	);

	const gridContainer = getByTestId("lyzer-grid-container");
	expect(gridContainer).toMatchSnapshot();
});

it("Grid container renders correctly with grid items", () => {
	const { getByTestId } = render(
		<GridContainer testId="lyzer-grid-container">
			<GridItem xs={12} sm={9} md={6} lg={4} xl={3} xxl={1}>Grid Item</GridItem>
		</GridContainer>
	);

	const gridContainer = getByTestId("lyzer-grid-container");
	expect(gridContainer).toMatchSnapshot();
});
