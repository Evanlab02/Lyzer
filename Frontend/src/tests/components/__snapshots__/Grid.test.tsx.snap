// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Grid container renders correctly 1`] = `
<div
  class="flex-grid-container"
  data-testid="lyzer-grid-container"
>
  Grid Container
</div>
`;

exports[`Grid container renders correctly with grid items 1`] = `
<div
  class="flex-grid-container"
  data-testid="lyzer-grid-container"
>
  <div
    class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-9 flex-grid-col-md-6 flex-grid-col-lg-4 flex-grid-col-xl-3 flex-grid-col-xxl-1"
  >
    Grid Item
  </div>
</div>
`;
